### PR TITLE
feat: add transfer detail modal and fix status display

### PR DESCRIPTION
**Pull request by JuaniRios**
Closes RAI-278

## What

Adds a detail modal to the transfer panel and fixes the Status column to display proper DTO statuses instead of raw CQRS event type names.

## Why

The transfer panel had no way to inspect individual transfers — you could only see the summary row. Operators need to see the full event timeline (tx hashes, Alpaca IDs, failure reasons) when debugging transfer issues.

Separately, the Status column was showing raw aggregate event names like `MintRequested` or `DepositedIntoRaindex` because the REST `/transfers` endpoint extracted status by string-splitting the latest event type. The DTO status enums (`EquityMintStatus`, `EquityRedemptionStatus`, `UsdcBridgeStatus`) existed and worked correctly in the WebSocket path but were never used by the REST endpoint.

### Transfer detail modal (`transfer-panel.svelte`)
- Info button (ⓘ) on each row opens a native `<dialog>` modal
- Fetches event history from `GET /transfers/{kind}/{id}/events`
- Renders events as a vertical timeline with color-coded dots
- Transaction hashes render as clickable Basescan links
- Alpaca transfer IDs and `TransferRef` variants displayed appropriately
- Failure reasons and `DetectionFailure` (Timeout/ApiError) highlighted in red
- Skips noisy fields like `attestation` (raw byte arrays)

### Status column fix
- **`src/api.rs`**: Replaced the SQL-based `transfers_endpoint` (which extracted status via `row.latest_event.split("::").last()`) with aggregate replay via `load_all_transfer_operations()`. Deleted `TransferSummaryRow` and the 90-line `load_transfer_summary` function with its complex `json_extract` SQL
- **`src/dashboard/transfer_loader.rs`**: Added `load_all_transfer_operations()` and `replay_all()` — loads every aggregate, replays events, calls `to_dto()` to produce proper DTO statuses. No time cutoff (unlike the WebSocket path's 24h window)
- **`src/dashboard/mod.rs`**: Made `transfer_loader` module `pub(crate)` so `api.rs` can use it
- **`crates/dto/src/transfer.rs`**: Added `started_at()` accessor to `TransferOperation` (mirrors existing `updated_at()`)

### Other dashboard improvements
- Replaced the old "Detail" column with separate **Asset** and **Amount** columns
- Added `humanizeStatus()` to title-case snake_case DTO statuses (`pending_confirmation` → `Pending Confirmation`)
- Added hover tooltip on Status column header showing the status flow for each transfer type (Mint, Redeem, USDC Bridge)
- Added `rejected` to `statusStyle()` so the detail modal colors `MintRejected`/`RedemptionRejected` events correctly

## Testing

- Existing `transfer_loader` tests (6 tests) continue to pass — they exercise aggregate replay and DTO conversion
- Existing DTO tests (16 tests) pass including serialization format assertions
- All 78 transfer-related tests pass
- No new tests for the dashboard UI (Svelte component)

## Anything else

- The endpoint now replays every transfer aggregate on each request instead of doing a single SQL query. This trades query performance for correctness. If transfer volume grows significantly, this could be optimized with a materialized view or caching layer.

## Release Notes

* **New Features**
  * Added event history modals to view detailed transfer and trade operation timelines with status updates, timestamps, and transaction details.
  * Improved transfer panel with clearer Asset and Amount columns.

* **Improvements**
  * Enhanced transfer and trade listing endpoints with better filtering, sorting, and event retrieval capabilities.

### CODE REVIEW

**Comment by JuaniRios** [comment]
* **#631** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/631?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#629** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/629?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#616** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/616?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#628** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/628?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#621** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/621?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#620** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/620?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/620?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#619** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#618** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/618?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#615** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/615?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#614** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/614?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#613** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/613?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#612** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/612?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#611** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/611?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#606** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/606?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#603** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/603?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#602** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/602?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#601** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=620).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/dashboard/transfer_loader.rs [hunk 1]
```diff
@@ -84,6 +84,92 @@ pub(crate) async fn load_transfers(pool: &SqlitePool) -> LoadedTransfers {
...
 }
 
+/// Load all transfer DTOs for the REST endpoint's paginated listing.
+///
+/// Unlike [`load_transfers`] (which partitions into active/recent with a
+/// 24h cutoff for the WebSocket), this returns every transfer without
+/// filtering.  The caller handles time-range filtering and pagination.
+pub(crate) async fn load_all_transfer_operations(
+    pool: &SqlitePool,
+    kind_filter: Option<&[&str]>,
```
Comment (0xgleb) [inline src/dashboard/transfer_loader.rs]: > is `&str` the best we can do here in terms of type modeling?

i had the above thought after just reading the function signature but looking at the body, the answer is definitely yes. stuff like `include("equity_mint")` and other manual string matches seem like major foot guns

File: src/api.rs [hunk 2]
```diff
@@ -630,169 +630,158 @@ async fn transfers_endpoint(
...
-
-    let mut all_entries: Vec<serde_json::Value> = Vec::new();
-
-    for (aggregate_type, kind_label, started_path) in &type_configs {
-        if let Some(ref kind_filter) = kinds
-            && !kind_filter.iter().any(|val| val == *kind_label)
-        {
-            continue;
-        }
+        .map(|val| val.split(',').map(str::trim).collect());
```
Comment (0xgleb) [inline src/api.rs]: imo all of this should be done by serde and/or whatever other trait impls rocket requires and then the endpoint should take typed args. that's one of the selling points of the framework - you just specify what args you need and what you'll return and then requests/responses are auto decoded/encoded. in this case we'd first need a better type than `&str` to represent a filter but i've mentioned that in another comment already

File: src/api.rs [hunk 3]
```diff
@@ -630,169 +630,158 @@ async fn transfers_endpoint(
...
+    pool: &State<SqlitePool>,
+    kind: &str,
+    aggregate_id: &str,
+) -> Option<Json<serde_json::Value>> {
+    let aggregate_type = match kind {
+        "equity_mint" => "TokenizedEquityMint",
+        "equity_redemption" => "EquityRedemption",
+        "usdc_bridge" => "UsdcRebalance",
+        _ => return None,
+    };
```
Comment (0xgleb) [inline src/api.rs]: once again i think we can do much better type modeling for both args and return type. this should be a simple exhaustive mapping from one enum to another (if needed, maybe we just take the enum we actually need as an arg to begin with) and then serde+rocket can handle cases lik `_ => return None` automatically

File: dashboard/src/lib/components/transfer-panel.svelte [hunk 4]
```diff
@@ -178,19 +186,98 @@
...
+    for (const [key, value] of Object.entries(payload)) {
+      if (isTimestampField(key) && typeof value === 'string') return value
+    }
+    return null
+  }
+
+  const detailFields = (payload: Record<string, unknown>): Array<[string, unknown]> =>
+    Object.entries(payload).filter(
+      ([key]) => !isTimestampField(key) && !SKIP_FIELDS.has(key),
+    )
```
Comment (0xgleb) [inline dashboard/src/lib/components/transfer-panel.svelte]: nitpick: maybe worth pulling out some of this logic into a lib module

**Review by 0xgleb — CHANGES_REQUESTED** [review]
the diff deleting `load_transfer_summary` sql query is sooooo nice to see. however, left some comments on what could be improved in terms of type modeling

File: src/dashboard/transfer_loader.rs (hunk 1 above)
Comment (JuaniRios) [reply src/dashboard/transfer_loader.rs]: Fixed in 7d0e1e6. Added `TransferKind` enum with `FromStr`/`Display` and `aggregate_type()` mapping. The loader now takes `Option<&[TransferKind]>` instead of `Option<&[&str]>`.

File: src/api.rs (hunk 2 above)
Comment (JuaniRios) [reply src/api.rs]: Fixed in 7d0e1e6. Added `TransferKindFilter` with Rocket's `FromFormField` for the comma-separated query param. Invalid kinds now fail at the framework level.

File: src/api.rs (hunk 3 above)
Comment (JuaniRios) [reply src/api.rs]: Fixed in 7d0e1e6. `transfer_events` now takes `TransferKind` as a typed path param via `FromParam`. The manual string match is gone -- Rocket returns 404 for invalid kinds automatically.

File: dashboard/src/lib/components/transfer-panel.svelte (hunk 4 above)
Comment (JuaniRios) [reply dashboard/src/lib/components/transfer-panel.svelte]: Fixed in 7d0e1e6. Extracted 10 helpers (`statusStyle`, `humanizeStatus`, `extractTimestamp`, etc.) to `$lib/transfer.ts`.

File: dashboard/src/lib/transfer.ts
Comment (0xgleb) [inline dashboard/src/lib/transfer.ts:1]: i think we need a corresponding test module

**Review by 0xgleb — CHANGES_REQUESTED** [review]
i think transfer.ts needs a corresponding test module with unit tests for all the logic in transfer.ts but aside from that looks great

File: dashboard/src/lib/transfer.ts
Comment (JuaniRios) [reply dashboard/src/lib/transfer.ts:1]: Added test module in `dashboard/src/lib/transfer.test.ts` covering `statusStyle` (priority ordering, case-insensitive matching, PascalCase events), `isTxHash` (boundary conditions, type guard), `extractTimestamp` (first-match-wins, type filtering), and `detailFields` (filter interaction, order preservation). Fixed in 1d308651.

**Review by 0xgleb — APPROVED** [review]

**Review by hardyjosh — APPROVED** [review]
